### ISSUE TITLE
[HOLD for payment 2024-12-17] [HOLD for payment 2024-12-16] [$250] Expense - Not here page shows up briefly when deleting the expense

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#45576]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.8-1
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**Found when executing PR :** https://github.com/Expensify/App/pull/44537
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to DM.
3. Submit two expenses to the user.
4. Open any of the transaction thread.
5. Tap on the report header.
6. Tap Delete expense..
7. Delete the expense.

## Expected Result:
Not here page will not show up when deleting the expense.

## Actual Result:
Not here page shows up briefly when deleting the expense.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/819b601e-65bc-4aab-82ea-69cfa7faa418

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c66dfe570a289fac</li>
        <li>Upwork Job ID: 1813534350182069816</li>
        <li>Last Price Increase: 2024-08-20</li>
<li>Automatic offers: </li>
<ul>
<li>alitoshmatov | Reviewer | 103693223</li>
<li>wildan-m | Contributor | 104145017</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @RachCHopkins</details>

**Comment by izarutskaya** [comment]
We think this issue might be related to the #collect project.

**Comment by izarutskaya** [comment]
Production 

https://github.com/user-attachments/assets/31cea64f-98a8-451a-8a98-802d893545f4

**Comment by trjExpensify** [comment]
@alitoshmatov can you review the proposal and keep this moving please? Thanks!

**Comment by alitoshmatov** [comment]
@daledah In https://github.com/Expensify/App/blob/963f8ce2c108d543c5058e7574f21465106440a6/src/pages/ReportDetailsPage.tsx#L694

We need `IOU.deleteMoneyRequest` to be executed so it returns a url to navigate back. How are planning on not calling this function but still getting url to navigate back?

**Comment by alitoshmatov** [comment]
@allgandalf Sorry missed your proposal. 

Your solution seems to be working but now, instead of not found page, title is changing into `[deleted expense]`

https://github.com/user-attachments/assets/53a2fef7-fb71-4685-b0ae-f0af7ea55a9c

<img width="459" alt="Screenshot 2024-08-10 at 7 25 11 PM" src="https://github.com/user-attachments/assets/f1d4046f-9b49-4480-8aaa-82c74fff03e8">

**Comment by sonialiap** [comment]
@RachCHopkins I'm OOO Aug 19-30, adding leave buddy. 
Status: waiting for proposals / C+ to review proposals

**Comment by alitoshmatov** [comment]
> That is because we wait for the action to get completed, but if we want it to happen in the background then we can do it without waiting for the interaction to be completed:

@allgandalf That does still not work

**Comment by alitoshmatov** [comment]
@daledah Can you provide detailed changes for your solution, I am still struggling to understand it.

**Comment by iwiznia** [comment]
No Melvin, still evaluating proposals

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by wildan-m** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

1. The "Not here" page briefly appears when deleting expenses.
2. [Deleted expense] briefly appear when deleting the expense

### What is the root cause of that problem?
1. We completely remove the thread report object that [cause shouldShowNotFoundPage true](https://github.com/Expensify/App/blob/f58439cfb116abd338dae880088a26725794108e/src/pages/home/ReportScreen.tsx#L425-L427). 
2. There is race condition between dismissing the `ReportDetailsPage` and deleting the expense.

These issues are noticeable on mobile devices due to their slower performance compared to web/desktop, but they occur across all platforms and layout sizes.

### What changes do you think we should make in order to solve the problem?
1. We can soft delete the report first and then completely delete the report object when the API call success. This pattern also implemented when we leaving a room.

https://github.com/Expensify/App/blob/f58439cfb116abd338dae880088a26725794108e/src/libs/actions/Report.ts#L2819-L2824

Change transaction thread optimistic data 

https://github.com/Expensify/App/blob/f58439cfb116abd338dae880088a26725794108e/src/libs/actions/IOU.ts#L5793-L5799

To

```tsx
    if (shouldDeleteTransactionThread) {
        optimisticData.push(
            {
                onyxMethod: Onyx.METHOD.MERGE,
                key: `${ONYXKEYS.COLLECTION.REPORT}${transactionThreadID}`,
                value: {
                    reportID: null,
                    stateNum: CONST.REPORT.STATE_NUM.APPROVED,
                    statusNum: CONST.REPORT.STATUS_NUM.CLOSED,
                    notificationPreference: CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN,
                },
            },
```

When success, remove remaining properties value to ensure the object completely deleted:

```tsx
    if(shouldDeleteTransactionThread) {
        successData.push({
            onyxMethod: Onyx.METHOD.MERGE,
            key: `${ONYXKEYS.COLLECTION.REPORT}${transactionThreadID}`,
            value: Object.keys(transactionThread as OnyxTypes.Report).reduce<Record<string, null>>((acc, key) => {
                        acc[key] = null;
                        return acc;
                    }, {}),
        })
    }
```

Do the same for `getDeleteTrackExpenseInformation`

2. To handle [deleted expense] issue, we can delay the delete process until the confirm modal hidden. Introduce new ref `isDeletedButtonPressed` to know that the deletion process confirmed.

Change

https://github.com/Expensify/App/blob/f58439cfb116abd338dae880088a26725794108e/src/pages/ReportDetailsPage.tsx#L788

To:

```tsx
                    onConfirm={()=>{
                        Navigation.dismissModal();
                        setIsDeleteModalVisible(false);
                        isDeletedButtonPressed.current = true;
                    }}
```

And modify `onModalHide` to:
```tsx
                    onModalHide={() => {
                        if (isDeletedButtonPressed.current) {
                            deleteTransaction();
                        }

                        // We use isTransactionDeleted to know if the modal hides because the user deletes the transaction.
                        if (!isTransactionDeleted.current) {
                            if (caseID === CASES.DEFAULT) {
                                if (navigateBackToAfterDelete.current) {
......
```

Remove `setIsDeleteModalVisible(false);` in `deleteTransaction`


[Branch for this solution](https://github.com/Expensify/App/compare/main...wildan-m:wildan/fix/45576-handle-not-found-delete-expense?expand=1)

### What alternative solutions did you explore? (Optional)
N/A
s

**Comment by alitoshmatov** [comment]
@wildan-m Thank you for your proposal, I really liked the first solution, but does not solve the complete problem. The second solution I think is similar to @daledah 's [proposal](https://github.com/Expensify/App/issues/45576#issuecomment-2233212003) in some way.

**Comment by alitoshmatov** [comment]
I think we can go with @daledah 's [proposal](https://github.com/Expensify/App/issues/45576#issuecomment-2233212003) which suggests delaying deletion until page animation is complete

C+ reviewed 🎀 👀 🎀

**Comment by wildan-m** [comment]
@alitoshmatov Thanks for your review. I wanted to clarify that I only proposed one solution for each issue. They should be implemented together since the root causes are different. My delay for the second issue is based on the hide modal event, not on a timeout. I believe this should be considered distinct.

**Comment by luacmartins** [comment]
Proposal LGTM and we already call the same `navigateBackAfterDeleteTransaction` in a few other places.

**Comment by RachCHopkins** [comment]
@alitoshmatov how is this going?

**Comment by luacmartins** [comment]
@daledah @alitoshmatov we reverted the PR associated with this issue because it caused the following regressions. Please take them into consideration when working on v2.

- https://github.com/Expensify/App/issues/49052
- https://github.com/Expensify/App/issues/49066
- https://github.com/Expensify/App/issues/49094

**Comment by alitoshmatov** [comment]
So looks like chosen solution was not a good one. We need to start over and choose a better proposal

Can we reapply `Help wanted` label to get some eyes
cc: @luacmartins @sonialiap

**Comment by alitoshmatov** [comment]
> @alitoshmatov Would you be able to take another look at my proposal? https://github.com/Expensify/App/issues/45576#issuecomment-2309556187 The branch link has been updated with the latest main, and my quick test doesn't show any regressions. (https://github.com/Expensify/App/issues/45576#issuecomment-2346844390)

Looks promising, let me do some more testing

**Comment by sonialiap** [comment]
@alitoshmatov how's the testing going?

**Comment by iwiznia** [comment]
Is there any progress here? It's been a week without update, should we re-assign?

**Comment by alitoshmatov** [comment]
I think we can go with @wildan-m's [proposal](https://github.com/Expensify/App/issues/45576#issuecomment-2309556187) which soft deletes transaction

C+ reviewed 🎀 👀 🎀

**Comment by wildan-m** [comment]
working on the PR, perform more tests

**Comment by wildan-m** [comment]
@alitoshmatov @iwiznia there is [an open issue](https://github.com/Expensify/App/issues/47584) that will block the result of the test of this issue in iOS, I can experience it in iOS, should we keep creating the PR or hold this one?

**Comment by iwiznia** [comment]
If you think that your code will work once that other issue is done, then let's send the PR but keep this issue on hold till we can actually test the results in all platforms.

**Comment by wildan-m** [comment]
@alitoshmatov @iwiznia after further testing, my solution doesn't solve all cases, the briefly [deleted] text also shown in the report header.

https://github.com/user-attachments/assets/08ca609f-2186-4ba4-8552-29467e65148c

While I search for another solution, you should consider reapplying the "help wanted" label to attract better proposals.

**Comment by alitoshmatov** [comment]
Thank you @wildan-m . Then we are back on choosing proposals

**Comment by alitoshmatov** [comment]
FYI: Looks like we have the same issue when deleting tasks

https://github.com/user-attachments/assets/b5f19406-1b5e-4ca7-b163-de43f9fd6a68

**Comment by wildan-m** [comment]
[Proposal Updated](https://github.com/Expensify/App/issues/45576#issuecomment-2309556187)

- Move prev main solution to alternative
- Add new main solution using decoupled navigation url generation

**Comment by wildan-m** [comment]
@alitoshmatov Any feedback about my updated proposal? I've noticed that I'm still assigned, if we are ok with the solution I'll create the PR.

**Comment by sonialiap** [comment]
@alitoshmatov bumping the request to [review proposal changes](https://github.com/Expensify/App/issues/45576#issuecomment-2401114914)

**Comment by luacmartins** [comment]
@alitoshmatov what's the latest here? Did you get a chance to look into [this proposal](https://github.com/Expensify/App/issues/45576#issuecomment-2309556187)?

**Comment by alitoshmatov** [comment]
Sorry for late response. @wildan-m Your proposal looks promising, can you update your solution with main branch, I think there are some changes there and it looks like effecting your solution

**Comment by wildan-m** [comment]
Resolving conflict with the latest main branch & re-testing...

**Comment by wildan-m** [comment]
[Proposal Updated](https://github.com/Expensify/App/issues/45576#issuecomment-2309556187)

- Resolve conflict with main
- Update branch

I can confirm that this iOS issue https://github.com/Expensify/App/issues/45576#issuecomment-2384466032 has resolved after https://github.com/Expensify/App/issues/47584 closed.

@alitoshmatov

**Comment by sonialiap** [comment]
@alitoshmatov bumping for proposal review

**Comment by alitoshmatov** [comment]
Okay I think we can go with @wildan-m 's [proposal](https://github.com/Expensify/App/issues/45576#issuecomment-2309556187)

C+ reviewed 🎀 👀 🎀

**Comment by iwiznia** [comment]
> In mWeb, even when we navigate to destination report earlier, not found might still appear,

Why would mWeb be different than mobile?

> navigateBackAfterDeleteTransaction

 Nitpick but let's rename this then since the order of operations is different now, maybe `navigateBackOnDeleteTransaction`

**Comment by wildan-m** [comment]
> Why would mWeb be different than mobile?

@iwiznia Using `Onyx.set` may be too quick for certain platforms or if React Navigation transition is slower than Onyx changes using set in Android mWeb. In such cases, we can opt for `Onyx.merge` instead. This approach is already utilized in [leaveRoom](https://github.com/Expensify/App/blob/04402e9f47a77f10b3c2c47fefe0931fbf62e4d4/src/libs/actions/Report.ts#L2958-L2967) and [leaveGroupChat](https://github.com/Expensify/App/blob/04402e9f47a77f10b3c2c47fefe0931fbf62e4d4/src/libs/actions/Report.ts#L2878-L2895) and it can effectively address similar issues.

**Comment by wildan-m** [comment]
PR creation blocked by this issue

`Cannot read properties of undefined (reading 'ownerAccountID')`

https://github.com/user-attachments/assets/1595b10c-32ab-4f40-8abf-9ba19f6e0423


Reported here 

https://expensify.slack.com/archives/C01GTK53T8Q/p1731569593270779

**Comment by iwiznia** [comment]
That should be fixed already @wildan-m

**Comment by wildan-m** [comment]
@iwiznia @alitoshmatov @sonialiap I can't reproduce the issue in the latest main, can you?

**Comment by wildan-m** [comment]
@alitoshmatov I can reproduce it again, especially in ios native. The PR is ready to review https://github.com/Expensify/App/pull/52740. Thanks!

**Comment by sonialiap** [comment]
I'm OOO Dec 16-20, adding a leave buddy

next steps:
- @alitoshmatov to complete [checklist](https://github.com/Expensify/App/issues/45576#issuecomment-2533125119) 
- BZ to make payment

**Comment by alitoshmatov** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] For any bug fix or new feature in this PR, I verified that sufficient [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) are included to prevent regressions in this flow.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by alitoshmatov** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] For any bug fix or new feature in this PR, I verified that sufficient [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) are included to prevent regressions in this flow.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by alitoshmatov** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [x] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [x] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: https://github.com/Expensify/App/pull/43251/files#r1887128320

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: No discussion started

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.


<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1. Open the App.
2. Go to any DM
3. Submit two expenses to the user.
4. Open any transaction thread.
5. Tap on the report header.
6. Select Delete Expense and confirm deletion.
7. Verify that the "Not here" page does not appear after deleting the expense. This could happen briefly, and usually in mobile devices

Do we agree 👍 or 👎

**Comment by alitoshmatov** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by RachCHopkins** [comment]
Offer sent @daledah https://www.upwork.com/nx/wm/offer/105394178!

**Comment by RachCHopkins** [comment]
Payment Summary:

- Contributor: @daledah  to be paid $250 via Upwork
- Contributor: @wildan-m  to be paid $250 via Upwork
- Contributor+: @alitoshmatov  to be paid $250 via Upwork

Upwork job [here](https://www.upwork.com/jobs/~01c66dfe570a289fac)

**Comment by RachCHopkins** [comment]
Contributors have all been paid, the contracts have been completed, and the Upwork post has been closed.

**Comment by wildan-m** [comment]
@RachCHopkins Just wanted to let you know that my proposal has replaced the previous contributor's proposal, I think the C+ for this issue only @alitoshmatov

**Comment by JmillsExpensify** [comment]
$250 approved for @alitoshmatov
